lib/buffer.js

```diff
@@ -370,8 +370,10 @@ Buffer.prototype.toString = function() {
   } else {
     var result = slowToString.apply(this, arguments);
   }
-  if (result === undefined)
-    throw new Error('toString failed');
+  if (result === undefined) {
+    throw new RangeError('toString() failed: buffer larger than maximum ' +
+                         'string size ' + binding.kStringMaxLength + ' bytes');
```

Comment:
I think it should be `string size of 268435440 bytes` – reads oddly without the `of`